### PR TITLE
feat!: remove `skipInstalls` config option

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- remove `skipInstalls` config optin from self-hosted configuration

## Context
- Closes: https://github.com/renovatebot/renovate/issues/10097

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
I don't understand the coverage fail. the line is ignored still the codecov asks for coverage.

**Review by viceice — DISMISSED** [review]
looks like a jest cache issue, do we have a test case for the uncovered line?

**Comment by RahulGautamSingh** [comment]
> looks like a jest cache issue, do we have a test case for the uncovered line?

No

> looks like a jest cache issue, do we have a test case for the uncovered line?

We don't have a test case, that's why we use `istanbul if` for it.

**Comment by rarkins** [comment]
@RahulGautamSingh needs conflict resolution

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -1294,7 +1286,7 @@ const options: RenovateOptions[] = [
     env: false,
   },
   {
-    name: 'matchFiles',
+    name: 'matchFileNames',
```
Comment (RahulGautamSingh) [inline lib/config/options/index.ts:1289]:
```suggestion
    name: 'matchFiles',
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -2210,6 +2210,20 @@ The following example matches any `package.json`, including files like `backend/
...
+  "packageRules": [
+    {
+      "description": "Group dependencies from package.json files",
+      "matchFileNames": ["**/package.json"],
+      "groupName": "All package.json changes"
+    }
+  ]
+}
+```
+
```
Comment (RahulGautamSingh) [inline docs/usage/configuration-options.md:2226]:
```suggestion
```
```

**Comment by rarkins** [comment]
@RahulGautamSingh looks like a bad merge - 27 files changed

**Review by rarkins — APPROVED** [review]

**Pull request merged**